core/src/smartcontracts/isi/domain.rs

```diff
@@ -77,7 +77,7 @@ pub mod isi {
             authority: <Account as Identifiable>::Id,
             wsv: &WorldStateView<W>,
         ) -> Result<(), Self::Error> {
-            let asset_definition = self.object;
+            let asset_definition = self.object.build();
```

Comment:
This raises a question. We have a lot of builders in ISI. Perhaps we need to formalise them in a `trait`. 


```rust
pub trait Builder {
    type Target;
    
    fn build(&self) -> Target;
}

Reply:
In addition to this being an extra sanity check, if we have a trait, at some point we'd be able to generify the registration process.

Reply:
#2253